src/StackExchange.Redis/PhysicalConnection.cs

```diff
@@ -875,11 +875,23 @@ internal static int WriteRaw(Span<byte> span, long value, bool withLengthPrefix
         [Obsolete("this is an anti-pattern; work to reduce reliance on this is in progress")]
         internal WriteResult FlushSync(bool throwOnFailure, int millisecondsTimeout)
         {
-            var flush = FlushAsync(throwOnFailure);
+            CancellationTokenSource source = new CancellationTokenSource();
```

Comment:
We need to wrap this in a using/dispose block

Reply:
good catch. fixed.